packages/web-frontend/src/containers/MapOverlayBar/TitleAndDatePicker.js

```diff
@@ -0,0 +1,157 @@
+import React, { useEffect, useState } from 'react';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+import styled from 'styled-components';
+import moment from 'moment';
+import { withStyles } from '@material-ui/core/styles';
+import DownArrow from '@material-ui/icons/ArrowDropDown';
+import MuiSwitch from '@material-ui/core/Switch';
+import CircularProgress from '@material-ui/core/CircularProgress';
+import { DateRangePicker } from '../../components/DateRangePicker';
+import { getDefaultDates, getLimits, GRANULARITY_CONFIG } from '../../utils/periodGranularities';
+import { Content, ContentText } from './Styles';
+import { TUPAIA_ORANGE } from '../../styles';
+import { setDisplayedMapOverlay } from '../../actions';
+
+const Switch = withStyles({
+  switchBase: {
+    '&$checked': {
+      color: TUPAIA_ORANGE,
+    },
+    '&$checked + $track': {
+      backgroundColor: TUPAIA_ORANGE,
+    },
+  },
+  checked: {},
+  track: {},
+})(MuiSwitch);
+
+const ToolsWrapper = styled.div`
+  display: flex;
+`;
+
+const IconWrapper = styled.div`
+  display: flex;
+  border-left: 1px solid rgba(255, 255, 255, 0.5);
+  padding: 8px 0 0 5px;
+`;
+
+const MeasureDatePicker = styled.div`
+  pointer-events: auto;
+  background: #203e5c;
+  padding: 16px 8px;
+  border-bottom-left-radius: ${({ expanded }) => (!expanded ? '5px' : '0')};
+  border-bottom-right-radius: ${({ expanded }) => (!expanded ? '5px' : '0')};
+`;
+
+export const TitleAndDatePickerComponent = ({
+  mapOverlay,
+  onUpdateMeasurePeriod,
+  isExpanded,
+  isMeasureSelected,
+  toggleMeasures,
+  isMeasureLoading,
+  displayedMapOverlays,
+  onSetDisplayedMapOverlay,
+  // TODO: PHX-103: show both date pickers
+  showDatePickerOnlyAfterSecondTitle,
+}) => {
+  const [isSwitchedOn, setIsSwitchedOn] = useState(true);
+  useEffect(() => {
+    setIsSwitchedOn(displayedMapOverlays.includes(mapOverlay?.mapOverlayId));
+  }, [JSON.stringify(displayedMapOverlays)]);
+  const switchOnChange = () => {
+    if (isSwitchedOn) {
+      onSetDisplayedMapOverlay(
+        displayedMapOverlays.filter(mapOverlayId => mapOverlayId !== mapOverlay?.mapOverlayId),
+      );
+    } else {
+      onSetDisplayedMapOverlay([...displayedMapOverlays, mapOverlay?.mapOverlayId]);
+    }
+    setIsSwitchedOn(!isSwitchedOn);
+  };
+  const { periodGranularity, isTimePeriodEditable = true, name } = mapOverlay;
```

Comment:
This is from the existing code:

```suggestion
  const { periodGranularity, isTimePeriodEditable = true, name } = mapOverlay || {};
```

You use `?.` with `mapOverlay`, so I assume that we can't be sure that it is actually defined.

Reply:
mapOverlay will always be an object in this component. I can assign `{}` to `mapOverlay` at props to make it clearer.

Reply:
That would be great!